epochStart/metachain/systemSCs.go

```diff
@@ -164,33 +127,35 @@ func NewSystemSCProcessor(args ArgsNewEpochStartSystemSCProcessing) (*systemSCPr
-		delegationEnableEpoch:       args.EpochConfig.EnableEpochs.DelegationSmartContractEnableEpoch,
-		stakingV2EnableEpoch:        args.EpochConfig.EnableEpochs.StakingV2EnableEpoch,
-		esdtEnableEpoch:             args.EpochConfig.EnableEpochs.ESDTEnableEpoch,
-		stakingDataProvider:         args.StakingDataProvider,
-		nodesConfigProvider:         args.NodesConfigProvider,
-		shardCoordinator:            args.ShardCoordinator,
-		correctLastUnJailEpoch:      args.EpochConfig.EnableEpochs.CorrectLastUnjailedEnableEpoch,
-		esdtOwnerAddressBytes:       args.ESDTOwnerAddressBytes,
-		saveJailedAlwaysEnableEpoch: args.EpochConfig.EnableEpochs.SaveJailedAlwaysEnableEpoch,
-		governanceEnableEpoch:       args.EpochConfig.EnableEpochs.GovernanceEnableEpoch,
-		builtInOnMetaEnableEpoch:    args.EpochConfig.EnableEpochs.BuiltInFunctionOnMetaEnableEpoch,
-		stakingV4InitEnableEpoch:    args.EpochConfig.EnableEpochs.StakingV4InitEnableEpoch,
-		stakingV4EnableEpoch:        args.EpochConfig.EnableEpochs.StakingV4EnableEpoch,
+		legacySystemSCProcessor: &legacySystemSCProcessor{
+			systemVM:                    args.SystemVM,
+			userAccountsDB:              args.UserAccountsDB,
+			peerAccountsDB:              args.PeerAccountsDB,
+			marshalizer:                 args.Marshalizer,
+			startRating:                 args.StartRating,
+			validatorInfoCreator:        args.ValidatorInfoCreator,
+			genesisNodesConfig:          args.GenesisNodesConfig,
+			endOfEpochCallerAddress:     args.EndOfEpochCallerAddress,
+			stakingSCAddress:            args.StakingSCAddress,
+			chanceComputer:              args.ChanceComputer,
+			mapNumSwitchedPerShard:      make(map[uint32]uint32),
+			mapNumSwitchablePerShard:    make(map[uint32]uint32),
+			switchEnableEpoch:           args.EpochConfig.EnableEpochs.SwitchJailWaitingEnableEpoch,
+			hystNodesEnableEpoch:        args.EpochConfig.EnableEpochs.SwitchHysteresisForMinNodesEnableEpoch,
+			delegationEnableEpoch:       args.EpochConfig.EnableEpochs.DelegationSmartContractEnableEpoch,
+			stakingV2EnableEpoch:        args.EpochConfig.EnableEpochs.StakingV2EnableEpoch,
+			esdtEnableEpoch:             args.EpochConfig.EnableEpochs.ESDTEnableEpoch,
+			stakingDataProvider:         args.StakingDataProvider,
+			nodesConfigProvider:         args.NodesConfigProvider,
+			shardCoordinator:            args.ShardCoordinator,
+			correctLastUnJailEpoch:      args.EpochConfig.EnableEpochs.CorrectLastUnjailedEnableEpoch,
+			esdtOwnerAddressBytes:       args.ESDTOwnerAddressBytes,
+			saveJailedAlwaysEnableEpoch: args.EpochConfig.EnableEpochs.SaveJailedAlwaysEnableEpoch,
+			stakingV4InitEnableEpoch:    args.EpochConfig.EnableEpochs.StakingV4InitEnableEpoch,
+		},
+		governanceEnableEpoch:    args.EpochConfig.EnableEpochs.GovernanceEnableEpoch,
+		builtInOnMetaEnableEpoch: args.EpochConfig.EnableEpochs.BuiltInFunctionOnMetaEnableEpoch,
+		stakingV4EnableEpoch:     args.EpochConfig.EnableEpochs.StakingV4EnableEpoch,
```

Comment:
[question]
Since we moved all subcomps and all legacy flags in `legacySystemSCProcessor`, should we create a constructor for it and move every nil check + `log.Debug("systemSC: enable epoch for....) ?
Or do we keep legacy just as a placeholder?

Reply:
split the code as suggested